From 2c001aa99eef9d65b2fa6ee86a07e77d89325d43 Mon Sep 17 00:00:00 2001
From: tpaviot <tpaviot@gmail.com>
Date: Thu, 28 Nov 2013 16:59:46 +0100
Subject: [PATCH] [osx-maverick-warning-fix][condition-always-true] Bugfix
 unreached code

Fix for following clang-5.0.0 warning on OSX 10.9:
/oce/src/AIS/AIS_TexturedShape.cxx:204:23: warning: comparison of constant -1 with expression of type 'Graphic3d_NameOfTexture2D' is always true [-Wtautological-constant-out-of-range-compare]
  if (myPredefTexture != -1)
      ~~~~~~~~~~~~~~~ ^  ~~
/Users/thomas/Devel/oce/src/AIS/AIS_TexturedShape.cxx:345:27: warning: comparison of constant -1 with expression of type 'Graphic3d_NameOfTexture2D' is always true [-Wtautological-constant-out-of-range-compare]
      if (myPredefTexture != -1)
          ~~~~~~~~~~~~~~~ ^  ~~
---
 src/AIS/AIS_TexturedShape.cxx | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

Index: oce/src/AIS/AIS_TexturedShape.cxx
===================================================================
--- oce.orig/src/AIS/AIS_TexturedShape.cxx
+++ oce/src/AIS/AIS_TexturedShape.cxx
@@ -85,7 +85,7 @@ void AIS_TexturedShape::SetTextureFileNa
   else
   {
     myTextureFile   = theTextureFileName;
-    myPredefTexture = Graphic3d_NameOfTexture2D (-1);
+    myPredefTexture = Graphic3d_NameOfTexture2D (Graphic3d_NOT_2D_UNKNOWN);
   }
 }
 
@@ -197,7 +197,7 @@ void AIS_TexturedShape::UpdateAttributes
     return;
   }
 
-  if (myPredefTexture != -1)
+  if (myPredefTexture != Graphic3d_NOT_2D_UNKNOWN)
     mytexture = new Graphic3d_Texture2Dmanual (myPredefTexture);
   else
     mytexture = new Graphic3d_Texture2Dmanual (myTextureFile.ToCString());
@@ -337,7 +337,7 @@ void AIS_TexturedShape::Compute (const H
       }
       myAspect->SetTextureMapOn();
 
-      if (myPredefTexture != -1)
+      if (myPredefTexture != Graphic3d_NOT_2D_UNKNOWN)
         mytexture = new Graphic3d_Texture2Dmanual (myPredefTexture);
       else
         mytexture = new Graphic3d_Texture2Dmanual (myTextureFile.ToCString());
